cmd/devenv/debug/problem_os.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 Outreach Corporation. All Rights Reserved.
+
+// Description: This file implements problem detectors for specific
+// OS related problems.
+
+package debug
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// ProblemLowDiskSpace detects if the underlying Docker daemon is
+// running low on disk space, wether it's in the VM or on the host.
+// https://github.com/getoutreach/devenv/wiki/LowDiskSpace
+//
+// TODO(jaredallard): This only applies to local clusters, we don't have
+// the ability to selectively run problems yet.
+var ProblemLowDiskSpace = Problem{
+	Resource:         "docker",
+	ID:               "LowDiskSpace",
+	ShortDescription: "Docker VM (or host) is running out of disk space",
+	Detector: func(ctx context.Context, _ runtime.Object) (string, bool, bool) {
+		cmd := exec.CommandContext(ctx, "docker", "run", "--rm", "busybox", "df", "-h", "/")
+		b, err := cmd.Output()
+		if err != nil {
+			var exitErr *exec.ExitError
+			if errors.As(err, &exitErr) {
+				return "failed to run 'df -h' though docker: " + string(exitErr.Stderr), false, true
+			}
+
+			return "failed to run df (no output)", false, true
+		}
+
+		// The output of df -h is:
+		// Filesystem                Size      Used Available Use% Mounted on
+		// overlay                  59.6G     13.9G     43.7G  24% /
+		//
+		// We want the second line, and the 5th column.
+
+		var used int
+		scanner := bufio.NewScanner(bytes.NewReader(b))
+		for i := 0; scanner.Scan() && i < 2; i++ {
+			// Skip the first line
+			if i == 0 {
+				continue
+			}
+
+			out := scanner.Text()
+			fields := strings.Fields(out)
+			if len(fields) < 6 {
+				return "failed to parse df output: " + out, false, true
+			}
+
+			usedStr := fields[4]
+			if !strings.HasSuffix(usedStr, "%") {
+				return "failed to parse df output: " + fields[4], false, true
```

Comment:
Nit: distinguish between this error and the previous error
```suggestion
				return "failed to parse df Use% output: " + fields[4], false, true
```